src/azure-cli-core/azure/cli/core/util.py

```diff
@@ -1360,3 +1360,9 @@ def should_encrypt_token_cache(cli_ctx):
     encrypt = cli_ctx.config.getboolean('core', 'encrypt_token_cache', fallback=fallback)
 
     return encrypt
+
+
+def is_autocomplete():
+    """Detect if the command is running in autocomplete mode"""
+    from knack.completion import ARGCOMPLETE_ENV_NAME
+    return os.environ.get(ARGCOMPLETE_ENV_NAME) is not None
```

Comment:
I think we already have similar logic: https://github.com/Azure/azure-cli/blob/5c57d1c88bb5a8daded21e5fb6dc933456e9026d/src/azure-cli-core/azure/cli/core/__init__.py#L74